### ISSUE TITLE
 camera parameters in final fits

### ISSUE THREAD

**Issue by ThibaultGROUEIX** [nkolot/SPIN#98]
Dear authors,

Is it possible to release to camera parameters predicted by the network in the final fits?
Thanks a lot,

-Thibault

**Comment by geopavlakos** [comment]
With our in-the-loop procedure, we optimize both the camera and the body shape using the network estimates as initialization. As a result, the camera parameters estimated by the network are not very meaningful anymore for the fits, so we don't store them in memory during training.

**Comment by ThibaultGROUEIX** [comment]
Hi Thanks for your kind reply.
I do not understand why the camera parameters estimated by the network are irrelevant.
 
I am trying to retrain SPIN with direct supervision,  directly from the final fits. It seems that I need the camera predictions to supervise this part of the model's output. 

Furthermore, how can the predicted SMPL models be reprojected in the input image without those camera parameters?

Sorry if these are dumb question, i am relatively new to pose estimation.

Best

**Comment by nkolot** [comment]
What you are saying makes sense, but in practice one does not need to directly supervise the camera parameter prediction. Instead, it is implicitly supervised through a reprojection loss. So the network learns to predict a set of camera parameters that minimizes the reprojection loss of the predicted 3D joints of the model.
Does this make sense?

**Comment by ThibaultGROUEIX** [comment]
Yes, it does, thanks.
If I recap what it means for training SPIN with direct supervision, I have two choices :
-- Keep the loss on 2D keypoints to implicitly train the camera parameters
-- Generate the final camera parameters of a pre-trained SPIN, then strongly supervise the predicted camera parameters

Do you think one option is better than the other?

**Comment by nkolot** [comment]
I haven't tried option (2). You need to be mindful about the fact that in the way that we represent the camera parameters [s, tx, ty], tx and ty are an order of magnitude or more smaller than the scale parameter s. So this might make direct parameter supervision tricky and will likely require some tuning. From my experience, I would suggest you to stick with option (1) since I've never seen issues with the camera prediction being off.

**Comment by ThibaultGROUEIX** [comment]
Awesome, thanks for sharing your XP! :)

**Comment by ThibaultGROUEIX** [comment]
I just noticed you actually ran the experiment (in supmat tab.4 ). Do you have the code ?
I trained a v1 for this but the performance is not great so far.
